packages/client-utils/src/sequence-manager.ts

```diff
@@ -0,0 +1,59 @@
+import type { BaseAccount } from './codegen/cosmos/auth/v1beta1/auth.js';
+
+/**
+ * @alpha
+ */
+export type TxSequencer = {
+  getAccountNumber: () => bigint;
+  getSequenceNumber: () => bigint;
+  resync: () => Promise<void>;
+};
+
+/**
+ * Manage sequence numbers for submitting transactions from a single address.
+ *
+ * @alpha
+ */
+export const makeTxSequencer = async (
+  fetchAccount: () => Promise<BaseAccount>,
+  { log = () => {} }: { log?: (...args: unknown[]) => void },
+): Promise<TxSequencer> => {
+  const initial = await fetchAccount();
+  const accountNumber: bigint = initial.accountNumber;
+  let sequenceNumber: bigint = initial.sequence;
```

Comment:
Suggestion for internal clarity:
```suggestion
  /** The sequence number to be included in the next transaction. */
  let nextSequenceNumber: bigint = initial.sequence;
```